inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -285,3 +286,52 @@ def refextract(obj, eng):
     elif len(text_references) > len(pdf_references):
         obj.log.info('Extracted %d references from text.', len(text_references))
         obj.data['references'] = text_references
+
+
+@with_debug_logging
+def set_document_type_and_refereed(obj, eng):
+    """Set the `refereed` field to the appropriate value based on the journals' `refereed`
+    and `proceedings` fields values. Also, modify the `document_type` field accordingly.
+
+    The journal information is retrieved using the `journal_record` field value which is a
+    json reference to the journal records.
+
+    Note:
+        If some of the journals present in the `publication_info` have no `journal_record`
+        json reference associated, only the ones that do have the associated reference
+        are taken into account in making the decisions. If there are no journals or none
+        have an associated json reference to the corresponding journal record, the function
+        return  None and no further changes are being made.
+
+    Args:
+        obj: a workflow object.
+        eng: a workflow engine.
+
+    Returns:
+        None
+
+    """
+    journals = replace_refs(get_value(obj.data, 'publication_info.journal_record'), 'db')
+
+    if not journals:
+        return None
+
+    if any(journal.get('refereed') and journal.get('proceedings') for journal in journals) \
+            and 'conference paper' not in obj.data.get('document_type'):
+                    obj.data['refereed'] = True
+
+    if any(journal.get('refereed') and not journal.get('proceedings') for journal in journals):
+        obj.data['refereed'] = True
+
+    if all(journal.get('proceedings') for journal in journals) \
+            and all(not journal.get('refereed') for journal in journals):
+        doc_types = obj.data.get('document_type')
+        for index, doc_type in enumerate(doc_types):
+            if doc_type == 'article':
+                del obj.data['document_type'][index]
+                if 'conference paper' not in doc_types:
+                    obj.data['document_type'].append('conference paper')
+                break
+
+    if all(not journal.get('refereed') for journal in journals):
```

Comment:
`journal.get('refereed', True)` to be sure that if any journal has no `refereed`, this flag doesn't get set.